cscs-checks/apps/cp2k/cp2k_check.py

```diff
@@ -5,58 +5,125 @@
 
 import reframe as rfm
 import reframe.utility.sanity as sn
-
-
-class Cp2kCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self):
+from hpctestlib.apps.cp2k import Cp2k
+
+
+REFERENCE_CPU_PERFORMANCE_SMALL = {
+    'dom:mc': {
+        'maint': (202.2, None, 0.05, 's'),
+        'prod': (202.2, None, 0.05, 's')
+    },
+    'daint:mc': {
+        'maint': (180.9, None, 0.08, 's'),
+        'prod': (180.9, None, 0.08, 's')
+    },
+    'eiger:mc': {
+        'maint': (70.0, None, 0.08, 's'),
+        'prod': (46.0, None, 0.05, 's')
+    },
+    'pilatus:mc': {
+        'maint': (70.0, None, 0.08, 's'),
+        'prod': (70.0, None, 0.08, 's')
+    },
+}
+
+REFERENCE_CPU_PERFORMANCE_LARGE = {
+    'daint:mc': {
+        'maint': (141.0, None, 0.05, 's'),
+        'prod': (113.0, None, 0.05, 's')
+    },
+    'eiger:mc': {
+        'maint': (46.0, None, 0.05, 's'),
+        'prod': (46.0, None, 0.05, 's')
+    },
+    'pilatus:mc': {
+        'maint': (46.0, None, 0.05, 's'),
+        'prod': (46.0, None, 0.05, 's')
+    },
+}
+
+REFERENCE_GPU_PERFORMANCE_SMALL = {
+    'dom:mc': {
+        'maint': (251.8, None, 0.15, 's'),
+        'prod': (240.0, None, 0.05, 's')
+    },
+    'daint:mc': {
+        'maint': (241.3, None, 0.05, 's'),
+        'prod': (241.3, None, 0.05, 's')
+    }
+}
+
+REFERENCE_GPU_PERFORMANCE_LARGE = {
+    'daint:mc': {
+        'maint': (199.6, None, 0.06, 's'),
+        'prod': (199.6, None, 0.06, 's')
+    }
+}
+
+
+class Cp2kCheck(Cp2k):
+    modules = ['CP2K']
+    executable = 'cp2k.psmp'
+    executable_opts = ['H2O-256.inp']
+    maintainers = ['LM']
+    tags = {'scs'}
+    strict_check = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
+        }
+    }
+    scale = parameter(['small', 'large'])
+    benchmark = parameter(['prod', 'maint'])
```

Comment:
I think that the `benchmark` should not be pointing to these tags. I think that this is a site implementation detail. They should be moved to the derived class.

Reply:
Nevermind. I have mistakenly confused this for the base class... 😬

Reply:
We must be consistent with the naming convention across the different apps. In amber, the `benchmark` parameter defines the benchmark name, and some other benchmark specific quantities (i.e. the scientific system). Extending this meaning of `benchmark` over to here, cp2k only has a single benchmark.

Probably makes sense to rename this parameter to something like `mode`?